tests/functional/u/unnecessary/unnecessary_dict_index_lookup.py

```diff
@@ -100,3 +100,10 @@ class Foo:
     for key_two, val_two in val.items():
         del outer_dict[key][key_two]  # [unnecessary-dict-index-lookup]
         break
+
+# Test partial unpacking of items
+# https://github.com/PyCQA/pylint/issues/5504
+
+d = {}
+for key, in d.items():
+    print(d[key])
```

Comment:
We should probably still raise an ``unnecessary-dict-index-lookup`` here but if that's too hard to do I'd merge a hot fix for the crash. 

```suggestion
for key, in d.items():
    print(d[key]) # [unnecessary-dict-index-lookup]
```

Reply:
Let me see if I follow. The dict index lookup is still necessary, because we don't have a `v` variable for the value, but you're saying you still want it to raise a msg? That would be more like `poor-unpacking-of-dict-items` more than `unnecessary-dict-index-lookup`, right? The index lookup is still needed.

Reply:
I checked the functional tests and we have something similar:
```python
d = {'key': 'value'}

for k, _ in d.items():
    d[k] += 'VALUE'
    if 'V' in d[k]:  # This is fine, if d[k] is replaced with _, the semantics change
        print('found V')
```
Where we do not raise a warning. So you're right it should be another message entirely to stay coherent. (Use .value() instead of item(), or something.)